Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -476,7 +476,7 @@ extension HTTPClient {
             self.lock.withLock {
                 if !cancelled {
                     cancelled = true
-                    channel?.pipeline.fireUserInboundEventTriggered(TaskCancelEvent())
+                    channel?.triggerUserOutboundEvent(TaskCancelEvent(), promise: nil)
```

Comment:
I think since cancellation is not guaranteed to do anything (it's best effort only), we can skip any errors, wdyt?

Reply:
@artemredkin firing through the `ChannelPipeline` should come out of the lock. So something like

```
let channel = self.lock.withLock {
    if !self.cancelled {
        self.cancelled = true
        return self.channel
    }
 }
channel?.triggerUserOutboundEvent(...)
```

Regarding ignoring the promise: I'm okay with that.

Reply:
fixed, thanks!